### PR TITLE
Ignore invalid (disconnected) segments while calculating lane routings

### LINKED ISSUE #1124: An error has occured in mod's OnBeforeSimulationTick method. Mod name: TM:PE 11.6.0 TEST
I have a save that I have been playing for several weeks. Yesterday evening, I went to sleep and when I started playing Cities Skylines today morning, I started getting this error. I tried using both TM:PE Stable and test. When I start a new city, I don't seem to get the error. When I disabled the mod, I don't get the error. Also when the game time is paused, the error doesn't appear. The game is unplayable because I get the error every 15-30 seconds. Also when you go to main menu, you get a second error, but that is only in Test version.

**_Complete error message:_** 
An error has occured in mod's OnBeforeSimulationTick method.
Mod name: TM:PE 11.6.0 TEST - Manage your city's traffic
Assembly: TrafficManager, Version=11.6.0.36640, Culture=neutral, PublicKeyToken=null
Severity: Medium

System.InvalidOperationException: Nullable object must have a value.
  at System.Nullable`1[System.Boolean].get_Value () [0x00000] in <filename unknown>:0 
  at TrafficManager.Manager.Impl.RoutingManager.RecalculateLaneEndRoutingData (UInt16 segmentId, Int32 laneIndex, UInt32 laneId, Boolean startNode) [0x00000] in <filename unknown>:0 
  at TrafficManager.Manager.Impl.RoutingManager+<>c__DisplayClass23_0.<RecalculateSegment>b__0 (UInt32 laneId, .NetLane+ lane, .Lane laneInfo, UInt16 segId, .NetSegment+ segment, Byte laneIndex) [0x00000] in <filename unknown>:0 
  at CitiesGameBridge.Service.NetService.IterateSegmentLanes (UInt16 segmentId, .NetSegment+ segment, GenericGameBridge.Service.NetSegmentLaneHandler handler) [0x00000] in <filename unknown>:0 
  at CitiesGameBridge.Service.NetService.IterateSegmentLanes (UInt16 segmentId, GenericGameBridge.Service.NetSegmentLaneHandler handler) [0x00000] in <filename unknown>:0 
  at TrafficManager.Manager.Impl.RoutingManager.RecalculateSegment (UInt16 segmentId) [0x00000] in <filename unknown>:0 
  at TrafficManager.Manager.Impl.RoutingManager.SimulationStep () [0x00000] in <filename unknown>:0 
  at TrafficManager.Lifecycle.ThreadingExtension.OnBeforeSimulationTick () [0x00000] in <filename unknown>:0 
  at HealkitMod.Patches.ThreadingWrapperOnBeforeSimulationTickPatch.Prefix (System.Collections.Generic.List`1 ___m_ThreadingExtensions) [0x00000] in <filename unknown>:0 

**_Second error:_** https://pastebin.com/Q87AzWeF

**_Steps to reproduce:_** Unknown

**_How to trigger:_**
1. Play the game on the save I included, I should pop when you start the time.
2. You should get the second error message when you go to main menu  after loading the save. (Only on TM:PE Test)

**_Log files:_**
https://ufile.io/rfpr5n0s
I don't have TMPE.log

**_Savegame:_**
https://steamcommunity.com/sharedfiles/filedetails/?id=2514934321

**_Screenshot:_**
https://ufile.io/dnx8p2w3

### PR DESCRIPTION
**Pull request by krzychu124**
- Ignore invalid (disconnected) segments while calculating lane routings
- update outdated version number

Fixes #1124

### CODE REVIEW

File: TLM/TLM/Manager/Impl/RoutingManager.cs [hunk 1]
```diff
@@ -460,8 +461,13 @@ public class RoutingManager
                         continue;
                     }
 
-                    bool start = (bool)Constants.ServiceFactory.NetService.IsStartNode(segId, nextNodeId);
-                    ExtSegmentEnd segEnd = segEndMan.ExtSegmentEnds[segEndMan.GetIndex(segId, start)];
+                    bool? start = Constants.ServiceFactory.NetService.IsStartNode(segId, nextNodeId);
+                    if (!start.HasValue) {
+                        Log.Error($"Segment with id: {segId} is not connected to the node {nextNodeId}");
+                        Debug.LogError($"TM:PE RecalculateLaneRoutings - Segment with id {segId} is not connected to the node {nextNodeId}");
```
Comment (kianzarrin) [inline TLM/TLM/Manager/Impl/RoutingManager.cs:467]: why this one is debug.logerror ?
Log.Error does print to game file but Debug.LogError does not write to TMPE.

File: TLM/TLM/Util/VersionUtil.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ public static class VersionUtil {
         // we could alternatively use BuildConfig.APPLICATION_VERSION because const values are evaluated at compile time.
         // but I have decided not to do this because I don't want this to happen automatically with a rebuild if
         // CS updates. these values should be changed manaually so as to force us to acknowledge that they have changed.
-        public const uint EXPECTED_GAME_VERSION_U = 188868624U;
+        public const uint EXPECTED_GAME_VERSION_U = 189262096U;
```
Comment (kianzarrin) [inline TLM/TLM/Util/VersionUtil.cs:32]: was the game recently updated again?

**Review by kianzarrin — APPROVED** [review]
LGTM

File: TLM/TLM/Util/VersionUtil.cs (hunk 2 above)
Comment (krzychu124) [reply TLM/TLM/Util/VersionUtil.cs:32]: No, I think I forgot to update that const last time 😄

File: TLM/TLM/Manager/Impl/RoutingManager.cs (hunk 1 above)
Comment (krzychu124) [reply TLM/TLM/Manager/Impl/RoutingManager.cs:467]: Hmm interesting, I'll check, I remember that Log.Error didn't write to the game log

Comment (krzychu124) [reply TLM/TLM/Manager/Impl/RoutingManager.cs:467]: That's what I thought, `Log.Error` does not write to the game log (TMPE logger only writes to the TMPE.log).

**Review by DaEgi01 — APPROVED** [review]

**Pull request merged**